blocks/data-cards/data-cards.js

```diff
@@ -25,6 +27,92 @@ function createRTEContainer(rteElements) {
   return rteContainer;
 }
 
+/* Carousel Starts here */
+/* eslint-disable no-bitwise */
+
+function arrowIcon(props) {
+  const icon = img();
+  icon.src = `${window.hlx.codeBasePath}/icons/${props}.svg`;
+  icon.alt = `${props}`;
+  icon.loading = 'lazy';
+  icon.dataset.iconName = `${props}`;
+  return icon;
+}
+
+function arrow(props) {
+  const container = p({ class: 'arrow-button-container' });
+  const anchor = button({ class: 'carousel-button' });
+  anchor.classList.add(`button-${props}`);
+  anchor.type = 'button';
+  anchor.append(arrowIcon(props));
+  container.append(anchor);
+  return container;
+}
+
+function moveDirection(itemWidth, option) {
+  const carouselItems = document.querySelector('.data-card-items');
+  if (option === '+') {
+    carouselItems.scrollLeft += itemWidth;
+  } else {
+    carouselItems.scrollLeft -= itemWidth;
+  }
+}
+
+function buttonEvents(nextBtn, prevBtn) {
+  const cardsContainer = document.querySelector('.data-card-items');
```

Comment:
avoid referencing `document` wherever possible and target `block`.

Reply:
Updated @thomsebastin.

Reply:
I still see document.query at multiple place Line#53, Line#62,#63, #64